lib/cartopy/io/shapereader.py

```diff
@@ -233,6 +234,25 @@ def records(self):
             yield Record(shape_record.shape, geometry_factory, attributes,
                          fields)
 
+    def close(self):
+        """
+        Close a shapefile and its siblings (.shx and .dbf).
+        """
+        if hasattr(self._reader, 'close'):
```

Comment:
These two lines are the only two that I can support:

```
if hasattr(self._reader, 'close'):
    self._reader.close()
```

The detail in the else block is working around the fact that pyshp / shapefile does not support ``close`` before v2 (which hasn't been released). Unfortunately, it is not sustainable for cartopy to bare the brunt of tools that haven't released/implemented behaviour yet.

Reply:
I agree with @pelson here.